test/functional/helpers.lua

```diff
@@ -938,10 +938,9 @@ function module.missing_provider(provider)
     --- @type string?
     local e = module.fn['provider#' .. provider .. '#Detect']()[2]
     return e ~= '' and e or false
-  elseif provider == 'python' or provider == 'python3' then
-    local py_major_version = (provider == 'python3' and 3 or 2)
+  elseif provider == 'python' then
     --- @type string?
-    local e = module.fn['provider#pythonx#Detect'](py_major_version)[2]
+    local e = module.exec_lua("return vim.provider.python.detect_by_module('neovim')[2]")
```

Comment:
Related to the comment on `runtime/lua/vim/provider/python.lua`.

Reply:
It is? I don't get it.